src/dmd/dtemplate.d

```diff
@@ -8108,8 +8109,13 @@ MATCH matchArg(TemplateParameter tp, Scope* sc, RootObject oarg, size_t i, Templ
         if (sa)
         {
             if ((cast(Dsymbol)sa).isAggregateDeclaration())
+            {
+                // Don't lose type qualifiers
```

Comment:
I'm curious why it lost the qualifiers in the first place.

Reply:
I don't want to merge this until I find out why there's a special case for aggregates (the bug doesn't appear for non-aggregates). I suspect the real flaw is elsewhere.

Reply:
`Dsymbols` doesn't have qualifiers and you get one as result of the `getDsymbol` call a little above.
We need to investigate why is it there and if it's possible to drop it, or just avoid getting a symbol if the type has any qualifier.